imap/http_jwt.c

```diff
@@ -318,6 +323,25 @@ static int parse_token(struct jwt *jwt, const char *in, size_t inlen)
     return 1;
 }
 
+static int validate_pkey_type(struct jwt *jwt, EVP_PKEY *pkey)
+{
+    if (!jwt->nid)
+        return 0;
+
+    if (jwt->nid == EVP_PKEY_base_id(pkey))
```

Comment:
Shouldn't this be in an `#else` branch, similar to how it is done in `read_public_key`? Otherwise the use of `ENV_PKEY_base_id` isn't guaranteed to be safe in OpenSSL v3?